src/services/EdFi.DataManagementService.Api/Program.cs

```diff
@@ -3,20 +3,43 @@
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
 
+using EdFi.DataManagementService.Api.Configuration;
 using EdFi.DataManagementService.Api.Infrastructure;
+using Microsoft.Extensions.Options;
 
 var builder = WebApplication.CreateBuilder(args);
 builder.AddServices();
 
 var app = builder.Build();
 
 app.UseMiddleware<LoggingMiddleware>();
+
+InjectInvalidConfigurationMiddleware(app);
+
 app.UseRouting();
 app.UseRateLimiter();
 app.MapRouteEndpoints();
 
 app.Run();
 
+
+void InjectInvalidConfigurationMiddleware(WebApplication app)
+{
+    try
+    {
+        // Accessing IOptions<T> forces validation
+#pragma warning disable S1481 // Unused local variables should be removed
+        var appSettings = app.Services.GetRequiredService<IOptions<AppSettings>>().Value;
+        var connectionStrings = app.Services.GetRequiredService<IOptions<ConnectionStrings>>().Value;
+#pragma warning restore S1481 // Unused local variables should be removed
+    }
+    catch (OptionsValidationException ex)
+    {
+        app.UseMiddleware<InvalidConfigurationMiddleware>(ex.Failures);
+    }
```

Comment:
I was stepping through the debugger, and I realized that catching the exception right here - before `app.Run()` executes - means that the web server is still running and all of the routes will still be active.

Reply:
All of the routes are still active so we don't get 404's. This middleware is supposed to set the response to 500 and short circuit the rest of the pipeline.